frontend2/src/components/EpisodeSwitcher.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { Fragment, useMemo } from "react";
+import { Listbox, Transition } from "@headlessui/react";
+import Icon from "./elements/Icon";
+import { useEpisodeId, useEpisodeList } from "../contexts/EpisodeContext";
+import { isPresent } from "../utils/utilTypes";
+import { useNavigate } from "react-router-dom";
+
+const EpisodeSwitcher: React.FC = () => {
```

Comment:
Looks like this file is the SelectMenu but styled/moved logic into from Header...why exactly did you choose to fully rewrite the SelectMenu skeleton here (as opposed to i.e. building a styled variant of SelectMenu)? Just want to make sure we're not adding boilerplate code 😎

Reply:
I moved it out b/c EpisodeSwitcher doesn't seem like it belongs as a selectmenu variant and I wanted to avoid bloat in selectmenu
SelectMenu is something that should be used mainly in forms imo

Reply:
Ok sounds good!